realtime/websocket/__init__.py

```diff
@@ -60,25 +57,17 @@ async def websocket_handler(websocket: WebSocket):
                         bq = s
 
                 # TODO: Update the default sample rate to be consistent across all plugins
-                tasks = [
-                    WebsocketInputStream(websocket, audio_metadata.get("input_sample_rate", 48000)).run(
-                        audio_stream=audio_input_q, message_stream=text_input_q, video_stream=video_input_q
-                    ),
-                    WebsocketOutputStream(websocket, audio_metadata.get("output_sample_rate", 48000)).run(
-                        audio_stream=aq, message_stream=tq, video_stream=vq, byte_stream=bq
-                    ),
-                ]
-
+                websocket_input_processor = WebsocketInputProcessor(audio_stream=audio_input_q, message_stream=text_input_q, video_stream=video_input_q)
+                websocket_output_processor = WebsocketOutputProcessor(audio_stream=aq, message_stream=tq, video_stream=vq, byte_stream=bq)
+                create_and_run_server(path, websocket_input_processor, websocket_output_processor)
+                tasks = [websocket_input_processor.run(), websocket_output_processor.run()]
                 await asyncio.gather(*tasks)
-            except asyncio.CancelledError:
-                logging.error("websocket: CancelledError")
+
+                await create_and_run_server(path, websocket_input_processor, websocket_output_processor)
```

Comment:
Why are there two awaits? The second await will never be called since the first await runs forever.